src/usage_metrics/models.py

```diff
@@ -507,3 +508,196 @@
     Column("remote_ip_full_location", String),
     Column("partition_key", String),
 )
+
+
+core_eel_hole_logs = Table(
+    "core_eel_hole_logs",
+    usage_metrics_metadata,
+    Column(
+        "insert_id",
+        String,
+        primary_key=True,
+        comment="A unique identifier for the log entry.",
+    ),
+    Column(
+        "timestamp",
+        DateTime,
+        comment="The time the event described by the log entry occurred.",
+    ),
+    Column(
+        "text_payload",
+        String,
+        comment="Data provided to the logger in a text format. For the viewer, this includes the redirects generated by a user when logging in.",
+    ),
+    Column(
+        "event",
+        Enum("search", "hit", "duckdb_preview", "duckdb_csv", "log_in"),
+        comment="What type of event the log is recording (e.g., exporting to CSV, searching for a record).",
+    ),
+    Column(
+        "query",
+        String,
+        comment="What did a user type into the search box? This logs periodically, so multiple partial search queries may be logged as someone is typing.",
+    ),
+    Column(
+        "log_in_query",
+        String,
+        comment="What was the user doing when they were motivated to log in? This mirrors query when the search prompted a log-in event.",
+    ),
+    Column(
+        "name",
+        String,
+        comment="The name of the PUDL table returned by a search query. Only populated for 'hit' event types.",
+    ),
+    Column(
+        "score",
+        Float,
+        comment="The table's relevance score based on the provided search query. Only populated for 'hit' event types.",
+    ),
+    Column(
+        "tags",
+        String,
+        comment="The tags associated with a given table in the search results. Only populated for 'hit' events types.",
+    ),
+    Column(
+        "url",
+        String,
+        comment="What endpoint is a user hitting?",
+    ),
+    Column(
+        "params_name",
+        String,
+        comment="The name of the table being queried using DuckDB.",
+    ),
+    Column(
+        "params_page",
+        Integer,
+        comment="The page of the query.",
+    ),
+    Column(
+        "params_per_page",
+        Integer,
+        comment="The number of records returned per DuckDB query. This is set by us, so it should be expected to hold constant without our intervention.",
+    ),
+    Column(
+        "params_filters_field_name",
```

Comment:
non-blocking but would it be tidier to have all of these params stored in a separate table?